bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/MetadataClientDriver.java

```diff
@@ -84,4 +84,23 @@ LedgerManagerFactory getLedgerManagerFactory()
     @Override
     void close();
 
+    /**
+     * Listener on listening the metadata client connection expiry event.
+     */
+    @FunctionalInterface
+    interface ConnectionExpiredListener {
+
+        /**
+         * Signal when connection is expired.
+         */
+        void onConnectionExpired();
```

Comment:
I would call this simply 'onFatalError' or something that is simply telling that service should suicide.

Session expiration, and maybe session, is a ZK concept.